File: www/mak/get_color.py
#!/usr/bin/env python3
import sys

# Your HTML content
body = """<!DOCTYPE html>
<html>
<head><title>CGI Demo</title></head>
<style>
	body {
		background: linear-gradient(135deg, #2d2d2d, #1a1a1a);
		color: #90EE90;
	}
</style>
<body>
  <h1>Hello CGI World!</h1>
  <p>This demonstrates HTTP headers in Python CGI</p>
  <h2>Hello h2</h2>
</body>
</html>"""

# Calculate content length in bytes
content_length = len(body.encode('utf-8'))

# Build headers
headers = [
    "HTTP/1.1 200 OK",
    "Content-Type: text/html",
    f"Content-Length: {content_length}",
    "Set-Cookie: theme=dark; Max-Age=30",
    ""  # Empty string for final CRLF
]

# Print headers and body
sys.stdout.write("\r\n".join(headers))
sys.stdout.write("\r\n" + body)

File: www/mak/get_color.php
<?php
    // Store the HTML content in a variable
    $htmlContent = '
		<!DOCTYPE html>
		<html lang="en">
		<head>
			<meta charset="UTF-8">
			<meta name="viewport" content="width=device-width, initial-scale=1.0">
			<title>Hello Page</title>
			<style>
				body {
					font-family: Arial, sans-serif;
					background-color: #f4f4f4;
					color: #333;
					display: flex;
					justify-content: center;
					align-items: center;
					height: 100vh;
					margin: 0;
				}
				h1 {
					color: #007bff;
					font-size: 13rem;
				}
			</style>
		</head>
		<body>
			<h1>Hello</h1>
		</body>
		</html>
    ';

    // Calculate the length of the HTML content in bytes
    $contentLength = strlen($htmlContent);

    // Output the headers manually
    echo "HTTP/1.1 200 OK\n";
    echo "Content-Type: text/html\n";
    echo "Content-Length: $contentLength\n\n"; // Two newlines to end headers

    // Output the HTML content
    echo $htmlContent;
?>
